### PR TITLE
Decouple Plugins

### PR DESCRIPTION
**Pull request by WidgetsBurritos**
Drupal Issue: https://www.drupal.org/node/2894731

This PR decouples plugins. It was a combination effort between @bighappyface and myself.

- c0f061b - Remove legacy fields and plugin forms		
- 6277b8d - Improve schema and forms		
- aba053e - Cleanup minor issues		
- 284b3af - Archive edit form		
- f07cee9 - Add screenshot capture fields		
- 046ae58 - Fix test failure		
- 15f6db3 - Fixes failing tests and coding standards issues		
- 603c359 - Save on utility deletion		
- 23f33f5 - Tweak screenshot capture form

### CODE REVIEW

File: config/schema/web_page_archive.schema.yml [hunk 1]
```diff
@@ -21,14 +21,37 @@ web_page_archive.web_page_archive.*:
...
+  type: mapping
+  label: 'Capture utility settings'
+
+web_page_archive.capture_utility.html_capture_utility:
+  type: mapping
+  label: 'HTML Capture utility'
+  mapping:
+    capture:
+      type: boolean
+      label: 'Capture?'
```
Comment (WidgetsBurritos) [inline config/schema/web_page_archive.schema.yml:46]: Do we need this boolean value here?  Isn't the presence of the utility sufficient in this context?

File: src/Form/CaptureUtilityFormBase.php [hunk 2]
```diff
@@ -39,6 +39,10 @@ public function getFormId() {
   /**
    * {@inheritdoc}
    *
+   * @param array $form
+   *   The form array.
+   * @param Drupal\Core\Form\FormStateInterface $form_state
```
Comment (WidgetsBurritos) [inline src/Form/CaptureUtilityFormBase.php]: For consistency you mind changing `Drupal\` to `\Drupal\` here?

File: src/Form/WebPageArchiveEditForm.php [hunk 3]
```diff
@@ -18,58 +17,60 @@
...
+  }
 
   /**
    * {@inheritdoc}
    */
-  public function form(array $form, FormStateInterface $form_state) {
-    $form = parent::form($form, $form_state);
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity.manager')->getStorage('web_page_archive'),
```
Comment (WidgetsBurritos) [inline src/Form/WebPageArchiveEditForm.php]: `entity.manager` is deprecated.  We should use `entity_type.manager` instead (standard operations) or `entity.repository` instead (loading by uuid).  I haven't looked to see which context makes sense yet.

File: src/Form/WebPageArchiveFormBase.php [hunk 4]
```diff
@@ -41,92 +41,50 @@ public function __construct(EntityStorageInterface $image_style_storage) {
    */
   public static function create(ContainerInterface $container) {
     return new static(
-      $container->get('entity.manager')->getStorage('image_style')
+      $container->get('entity.manager')->getStorage('web_page_archive')
```
Comment (WidgetsBurritos) [inline src/Form/WebPageArchiveFormBase.php]: entity_type.manager here too

File: src/Plugin/CaptureUtility/ScreenshotCaptureUtility.php [hunk 5]
```diff
@@ -83,66 +82,29 @@ public function missingDependencies() {
...
-    //   // '#default_value' => $config[$this->pluginId],
-    //   // TODO: How to determine this?
-    //   '#default_value' => TRUE,
-    // ];
     $form['width'] = [
       '#type' => 'number',
       '#title' => $this->t('Capture width (in pixels)'),
       '#description' => $this->t('Specify the width you would like to capture.'),
-      // TODO: Get this value...
       '#default_value' => 1280,
```
Comment (WidgetsBurritos) [inline src/Plugin/CaptureUtility/ScreenshotCaptureUtility.php]: I think we should leave this todo in here for now.  If we edit a capture plugin, we're gonna want to retain the previously set value and this todo serves as a reminder to fix that.

File: src/Form/WebPageArchiveEditForm.php (hunk 3 above)
Comment (WidgetsBurritos) [reply src/Form/WebPageArchiveEditForm.php]: Actually on second thought.  Let's just leave it for now.  There are other instances of entity.manager in the code that will need to be converted as well, so we can address that all in a separate issue.  I'll update the active issues.

File: config/schema/web_page_archive.schema.yml (hunk 1 above)
Comment (bighappyface) [reply config/schema/web_page_archive.schema.yml:46]: It maps to the checkbox added by `HTMLCaptureUtility:: buildConfigurationForm`

I am cool to drop it, but this is how I directly linked the form element to the schema field.

**Comment by WidgetsBurritos** [comment]
@bighappyface,

We probably want to pull someone else in for the cumulative review here, but for your individual commits: 
+1 on 6277b8d
+1 on aba053e
+1 on 284b3af
+1 on f07cee9
+1 on 046ae58

**Review by bighappyface — APPROVED** [review]

**Pull request merged**